### ISSUE TITLE
Error: Strange wp-config.php file: wp-settings.php is not loaded directly.

### ISSUE THREAD

**Issue by raggascum** [wp-cli/wp-cli#4797]
I just installed CLI and here's my wp --info

```
PHP binary:	/usr/bin/php-7.1
PHP version:	7.1.15
php.ini used:	/etc/php.ini
WP-CLI root dir:	phar://wp-cli.phar
WP-CLI vendor dir:	phar://wp-cli.phar/vendor
WP_CLI phar path:	/usr/local
WP-CLI packages dir:	
WP-CLI global config:	
WP-CLI project config:	
WP-CLI version:	1.5.1
```

When I type in a wp command like wp config I get the following:
```
PHP Warning:  file_get_contents(/var/www/html/wordpress/wp-config.php): failed to open stream: Permission denied in phar:///usr/local/bin/wp/php/WP_CLI/Runner.php on line 552
Error: Strange wp-config.php file: wp-settings.php is not loaded directly.

```
I'm not sure what's going on here since this is the only wp-config file and the define for wp-settings is there.

**Comment by raggascum** [comment]
I have and there's been no response. I see people asking for help here often so why now is it a big issue? If you know how to help can't you just help instead of trying to display your dominance?

**Comment by antongorodezkiy** [comment]
@schlessera 

I had this issue when the code in `wp-config.php` file looked like this:
```
[...]
/** Sets up WordPress vars and included files. */\r\rrequire_once(ABSPATH . 'wp-settings.php');
```
Note: `\r\r` was invisible in my atom editor but I could see those symbols by using `sed -n 'l' wp-config.php` (do not ask why the code was formatted like this. it wasn't my code)

I think it's an issue, because wordpress actually worked with that code so PHP interpret it as correct.

**Comment by thomasbennett** [comment]
Probably need to adjust your wp-config.php to fix a bug/typo here. Make this code exists at the bottom:

```
/** Absolute path to the WordPress directory. */
if ( !defined('ABSPATH') )
    define('ABSPATH', dirname(__FILE__) . '/');

/** Sets up WordPress vars and included files. */
require_once ABSPATH . 'wp-settings.php';
```

**Comment by tovabbi** [comment]
Since google brings here, I would post also the solution: run wp-cli as www-data user https://www.alexgeorgiou.gr/wp-cli-www-data-user-permissions-linux/

**Comment by helgatheviking** [comment]
Google _still_ brings you here as these types of permissions issues present a serious barrier to the average developer/user who is not super well-versed in linux server admin stuff.  If you follow the install instructions and then run into this error, but have no idea how to fix it... `wp-cli` becomes a tool you can't use. 

It's understandable that the project can't account for every person's set up, but an issue that came up in 2018 and is still happening in 2021 with multiple people on this thread would suggest to me that this is recurring enough to provide some docu for. 

I've clicked on the www-data permissions link above and find it a little hard to follow, but if that's the solution it should be documented here (and maybe clarified a bit) in case anything happens to that site.

**Comment by Danrancan** [comment]
Agreed, this needs to be fixed. It's an issue that has been going on for years now. Why is nobody doing anything about this?

**Comment by johnycage** [comment]
Agreed

> Google _still_ brings you here as these types of permissions issues present a serious barrier to the average developer/user who is not super well-versed in linux server admin stuff. If you follow the install instructions and then run into this error, but have no idea how to fix it... `wp-cli` becomes a tool you can't use.
> 
> It's understandable that the project can't account for every person's set up, but an issue that came up in 2018 and is still happening in 2021 with multiple people on this thread would suggest to me that this is recurring enough to provide some docu for.
> 
> I've clicked on the www-data permissions link above and find it a little hard to follow, but if that's the solution it should be documented here (and maybe clarified a bit) in case anything happens to that site.

Agreed. Cannot be marked as "closed"

**Comment by Danrancan** [comment]
Can you give an explanation of why this issue is closed?

**Comment by JackBortoneLab** [comment]
For security reasons i want to move away wp-config.php from the document root so i rename wp-config.php as /etc/site/.env.php and define ABSPATH to '/home/www/mysite' in the file. Also permissions for this file can be restricted. Anyway thanks for reporting. I'll be looking for another way to modify a simple db value in my dev database until this issue is fixed.

**Comment by mastrobardo** [comment]
this should not be closed. I got into the same error 5 years later

**Comment by helgatheviking** [comment]
Hey google bringing me back in 2024! Nice to see everybody here again. :/

**Comment by helgatheviking** [comment]
Updating! I have some other site issues sorta convoluting things, BUT I have been able to get my CLI commands working finally.

After SSHing into my server as my personal user and navigating to the root of my WordPress install, I ran @antongorodezkiy's command `sed -n 'l' wp-config.php` and saw that my `wp-config.php` had a lot of rogue `/r` characters. So I copied the file locally. 

Then I removed the file entirely
`rm wp-config.php`

Then I copied `wp-config-sample.php` to `wp-config.php` to get a fresh start
`cp wp-config-sample.php wp-config.php`

So might need `sudo` for those. Then I went to edit the new `wp-config.php` and put all my DB values back in.
`sudo nano wp-config.php`

Then I had to give ownership of the config to `www-data`

`sudo chown www-data:www-data wp-config.php`

And now `wp help` will actually list the commands versus give me errors.
